packages/fxa-auth-server/lib/server.js

```diff
@@ -238,6 +239,40 @@ async function create(log, error, config, routes, db, translator, statsd) {
       return db.devices(uid);
     });
 
+    defineLazyGetter(request.app, 'isMetricsEnabled', async () => {
+      // This catches most but not all cases where the given uid
+      // is opted out and saves us from making a db call further down.
+      // Note that unverified accounts can not be opted out of metrics.
+      let uid;
+      if (
+        request.auth &&
+        request.auth.credentials &&
+        request.auth.credentials.uid
+      ) {
+        // sessionToken strategy comes with uid as uid
+        uid = request.auth.credentials.uid;
+      } else if (
+        request.auth &&
+        request.auth.credentials &&
+        request.auth.credentials.user
+      ) {
+        // oauthToken strategy comes with uid as user
+        uid = request.auth.credentials.user;
```

Comment:
`metricsOptOutAt` will be set on `credentials` in the sessionToken case, and _i think_ in the oauth case too

Reply:
actually it doesn't look like oauth has that. we might want to add it? 🤔